vignettes/adtte.Rmd

```diff
@@ -0,0 +1,272 @@
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(admiralonco)
+library(dplyr)
+library(admiral.test)
+```
+
+```{r, warning=FALSE, message=FALSE, include=FALSE}
+library(lubridate)
+```
+
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Derive Parameters (`CNSR`, `ADT`, `STARTDT`)](#parameters)
+* [Derive Analysis Value (`AVAL`)](#aval)
+* [Derive Analysis Sequence Number (`ASEQ`)](#aseq)
+* [Add ADSL Variables](#adslvars)
+
+## Read in Data {#readdata}
+
+To start, all datasets needed for the creation of the time-to-event dataset
+should be read into the environment. This will be a company specific process.
+
+For example purpose, the ADaM datasets---which are included
+in `{admiral}` and `{admiralonco}`---are used. An alternative might be to use
+`ADEVENT` as input.
+
+```{r}
+data("admiral_adsl")
+data("admiral_adrs")
+adsl <- admiral_adsl
+adrs <- admiral_adrs
+```
+
+## Derive Parameters (`CNSR`, `ADT`, `STARTDT`) {#parameters}
+
+To derive the parameter dependent variables like `CNSR`, `ADT`, `STARTDT`,
+`EVNTDESC`, `SRCDOM`, `PARAMCD`, ... the `admiral::derive_param_tte()` function can be
+used. It adds one parameter to the input dataset with one observation per
+subject. Usually it is called several times.
+
+For each subject it is determined if an event occurred. In the affirmative the
+analysis date `ADT` is set to the earliest event date. If no event occurred, the
+analysis date is set to the latest censoring date.
+
+The events and censorings are defined by the `admiral::event_source()` and the
+`admiral::censor_source()` class respectively. It defines
+
+- which observations (`filter` parameter) of a source dataset (`dataset_name`
+parameter) are potential events or censorings,
+- the value of the `CNSR` variable (`censor` parameter), and
+- which variable provides the date (`date` parameter).
+
+The date can be provided as date (`--DT` variable), datetime (`--DTM` variable),
+or character ISO-8601 date (`--DTC` variable).
+
+CDISC strongly recommends `CNSR = 0` for events and positive integers for
+censorings. `{admiral}`/`{admiralonco}` enforce this recommendation. Therefore the `censor`
+parameter is available for `admiral::censor_source()` only. It is defaulted to `1`.
+
+The `dataset_name` parameter expects a character value which is used as an
+identifier. The actual data which is used for the derivation of the parameter is
+provided via the `source_datasets` parameter of `admiral::derive_param_tte()`. It expects
+a named list of datasets. The names correspond to the identifiers specified for
+the `dataset_name` parameter. This allows to define events and censoring
+independent of the data.
+
+### Pre-Defined Time-to-Event Source Objects
+
+The table below shows all pre-defined `tte_source` objects which should cover the most
+common oncology use cases.
+
+```{r echo=FALSE}
+knitr::kable(list_tte_source_objects(package = "admiralonco"))
+```
+
```

Comment:
I wonder if we should remove `ANL01FL == "Y"` from the conditions of the `tte_source` objects. For parameters like PD, DEATH and LASTA it does not make much sense as there is just one observation per subject, i.e., it is fulfilled anyway.

Reply:
technically you're right but i prefer to keep them in for consistency and reminding of this important consideration of the ADRS workflow, plus user could add ANL02FL / ANLxxFL etc with different scenarios or for periods so it would remind them to consider this

Reply:
If you have `ANL01FL` for selecting the worst assessment per date and `ANL02FL` for selecting assessments prior anti cancer therapy for the OVR parameter, would you set both variables for other parameters like PD, RSP, or DEATH? And would you always use both variables when selecting observations for PD, RSP, or DEATH?

Reply:
i think in that scenario users would probably create new parameters - e.g. you'd have a PARAMCD=PD with ANL01FL=Y record and also say a PARAMCD=PDNACT with ANL02FL=Y. as you could have a patient who is a PD in the first condition but not in the 2nd, as the PD occurred after NACT (next anti cancer therapy). again your point would still stand as the ANLxxFLs wouldn't add anything to these records, except it just gives a more visual tie to the source records that contributed to each parameter, so would be appreciated i'm sure by an FDA reviewer.

the other more complex scenario (but likely rare) would be efficacy by period/phase - where you might have one PD record but ANLxxFL would get set dependent on which period/phase the event occurred at. in this case if you then needed to downstream do say a "time to event during neo-adjuvant phase" output then filtering with the right ANLxxFL would be critical.

in a nutshell, i agree with your comment but i rather leave it in as it does little harm and it opens up to users to consider possibly more complex protocol-specific needs.